ethyca/fideslog
# Write to S3

Closes #10

* [x] Include `boto3`
* [x] Instantiate `boto3` session
* [x] Replace db writing with dumping to s3
* [x] Replace test location with proper config
* [x] Update deployment and add testing (as required)

To test:
1.modify the endpoint of the SDK to use localhost instead
1. install locally in a virtual env
1. write some fake events (example below)
```
from datetime import datetime, timezone

from fideslog.sdk.python.client import AnalyticsClient
from fideslog.sdk.python.event import AnalyticsEvent
from fideslog.sdk.python.exceptions import AnalyticsError

try:
    client_id = "internal"
    os = "darwin"
    product_name = "fideslog"
    production_version = "1.1.1"
    analytics_client = AnalyticsClient(client_id, os, product_name, production_version)
    event = "test_event"
    fideslog_event = AnalyticsEvent(
        event="test_event",
        event_created_at=datetime.now(timezone.utc),
        flags=["--dry", "--local"],
        extra_data={"some_key": "some_value"},
    )
    analytics_client.send(fideslog_event)
except AnalyticsError:
    pass
```

You should then be able to see the events you are creating listed in the s3 bucket
